### PR TITLE
Added different protocols to be allowed as asset base_url

### LINKED ISSUE #28238: file:// is not a valid url for asset embedding
**Symfony version(s) affected**: 3.4.12 (think currently all published symfony versions affected)

**Description**  

I want to use `file://` as `base_urls` in the `framework` `asset` configuration as this is a lot faster when generating pdfs from html.

**How to reproduce**  

`config.yml`:

```yaml
framework:
    assets:
        packages:
            pdf:
                base_urls: 'file://%kernel.project_dir%/web'
```

Twig:

```twig
{{ asset('test.jpg', 'pdf') }}
```

Ends up in:

```
(1/3) InvalidArgumentException"file:///Users/alex/development/symfony.lo/web" is not a valid URL
```

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no    
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | fixes #28238
| License       | MIT
| Doc PR        | symfony/symfony-docs#10347

In some cases you want to use the `file://` as base_url when you are for example generating pdf and want to avoid network requests to improve the pdf generation performance:

```yaml
framework:
    assets:
        packages:
            pdf:
                base_url: "file://%kernel.project_dir%/public"
```

usage:

```twig
{{ asset('image.jpg', 'pdf' }}
```

### CODE REVIEW

File: src/Symfony/Component/Asset/UrlPackage.php [hunk 1]
```diff
@@ -129,7 +129,7 @@ private function getSslUrls($urls)
         foreach ($urls as $url) {
             if ('https://' === substr($url, 0, 8) || '//' === substr($url, 0, 2)) {
                 $sslUrls[] = $url;
-            } elseif ('http://' !== substr($url, 0, 7)) {
+            } elseif (false === strpos($url, '://')) {
```
Comment (iltar) [inline src/Symfony/Component/Asset/UrlPackage.php]: If I have `://` in my query string, this will now return true, can you add a testcase for that?

Comment (alexander-schranz) [reply src/Symfony/Component/Asset/UrlPackage.php]: @iltar fixed

File: src/Symfony/Component/Asset/UrlPackage.php [hunk 2]
```diff
@@ -129,7 +129,7 @@ private function getSslUrls($urls)
         foreach ($urls as $url) {
             if ('https://' === substr($url, 0, 8) || '//' === substr($url, 0, 2)) {
```
[the remark this answers is not in the document]
Comment (alexander-schranz) [reply src/Symfony/Component/Asset/UrlPackage.php]: I'm not sure if only relevant browser (http) secure protocols should listed here. If needed I can add sftp.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @alexander-schranz.

**Pull request merged**